### ISSUE TITLE
Frames do not implicitly end with \par

### ISSUE THREAD

**Issue by josephwright** [josephwright/beamer#227]
Originally reported by: **cyberSingularity (Bitbucket: [cyberSingularity](https://bitbucket.org/cyberSingularity), GitHub: Unknown)**

----------------------------------------

Not sure if you'll agree that this is a bug or not. But, I would expect the following two frames to look the same. Instead, the first is not centered because there is no explicit `\par` token/empty line at the end of the frame.

```
#!tex

\documentclass{beamer}
\begin{document}
	\begin{frame}\centering
		Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut purus elit,
		vestibulum ut, placerat ac, adipiscing vitae, felis. Curabitur dictum gravida
		mauris.
	\end{frame}
	
	\begin{frame}\centering
		Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut purus elit,
		vestibulum ut, placerat ac, adipiscing vitae, felis. Curabitur dictum gravida
		mauris.
		
	\end{frame}
	
\end{document}
```

Compare with the `article` class where no blank line is needed before `\end{document}` for `\centering` to take effect.

```
#!tex

\documentclass{article}
\begin{document}
	\centering
	Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Ut purus elit,
	vestibulum ut, placerat ac, adipiscing vitae, felis. Curabitur dictum gravida
	mauris.
\end{document}
```

Note that there may be other side-effects of the missing `\par` than just ignoring alignment for the last 'paragraph' in a `frame`.

----------------------------------------
- Bitbucket: https://bitbucket.org/rivanvx/beamer/issue/227

**Comment by josephwright** [comment]
*Original comment by* **Joseph Wright (Bitbucket: [josephwright](https://bitbucket.org/josephwright), GitHub: [josephwright](https://github.com/josephwright))**:

----------------------------------------

Probably true, but I wonder if altering the behaviour here could cause all sorts of problems.

**Comment by josephwright** [comment]
*Original comment by* **Joseph Wright (Bitbucket: [josephwright](https://bitbucket.org/josephwright), GitHub: [josephwright](https://github.com/josephwright))**:

----------------------------------------

Add \par at end of frame content (fixes #227)

As noted in bug report, there should be a `\par` here: with my 'LaTeX3 hat' on,
I note that we do much the same in vertical coffins for the same reason (TeX
does not use \ar-the-user-token here).
